[cppzmq] build failure - download hash matches but is wrong case

**Host Environment**
- OS: Windows

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install cppzmq`

**Failure logs** 

    Starting package 1/1: cppzmq:x86-windows
    Building package cppzmq[core]:x86-windows...
    CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:73 (message):

    File does not have expected hash:

          File path: [ C:/tools/vcpkg/downloads/zeromq-cppzmq-v4.8.1.tar.gz ]
      Expected hash: [ 02F9B77F67DD46557705511195EB3F4F4E52381256BC9687F36D3E69DB6A628C19CFFF02209B6E6B53822A60781AB0850EB064D8F020E059FC1ACA4D191B66DB ]
        Actual hash: [ 02f9b77f67dd46557705511195eb3f4f4e52381256bc9687f36d3e69db6a628c19cfff02209b6e6b53822a60781ab0850eb064d8f020e059fc1aca4d191b66db ]

    The cached file SHA512 doesn't match.  The file may have been corrupted.
    To re-download this file please delete cached file at path
    C:/tools/vcpkg/downloads/zeromq-cppzmq-v4.8.1.tar.gz and retry.

**Additional context**
Notice that the hashes are the same, only the expected is uppercase and the actual is lowercase. Hash matching should be case insensitive.

--- a/scripts/cmake/vcpkg_download_distfile.cmake
+++ b/scripts/cmake/vcpkg_download_distfile.cmake
@@ -69,7 +69,8 @@ function(z_vcpkg_download_distfile_test_hash file_path kind error_advice sha512 
     endif()
 
     file(SHA512 "${file_path}" file_hash)
-    if(NOT "${file_hash}" STREQUAL "${sha512}")
+    string(TOLOWER "${sha512}" sha512_lower)
+    if(NOT "${file_hash}" STREQUAL "${sha512_lower}")
         message(FATAL_ERROR
             "\nFile does not have expected hash:\n"
             "        File path: [ ${file_path} ]\n"

